opensrp-pnc/src/main/java/org/smartregister/pnc/utils/PncUtils.java

```diff
@@ -376,4 +378,201 @@ public static void processEvents(@NonNull List<PncEventClient> pncEventClients)
+            }
+
+            if (detailMap.containsKey(PncConstants.FormGlobalConstants.HIV_STATUS_PREVIOUS)) {
+                defaultGlobal.put(PncConstants.FormGlobalConstants.HIV_STATUS_PREVIOUS, detailMap.get(PncConstants.FormGlobalConstants.HIV_STATUS_PREVIOUS));
+            }
+
+            if (detailMap.containsKey(PncConstants.FormGlobalConstants.HIV_STATUS_CURRENT)) {
+                defaultGlobal.put(PncConstants.FormGlobalConstants.HIV_STATUS_CURRENT, detailMap.get(PncConstants.FormGlobalConstants.HIV_STATUS_CURRENT));
+            }
+
+            if (detailMap.containsKey(PncConstants.FormGlobalConstants.BABY_COMPLICATIONS)) {
+                defaultGlobal.put(PncConstants.FormGlobalConstants.BABY_COMPLICATIONS, detailMap.get(PncConstants.FormGlobalConstants.BABY_COMPLICATIONS));
+            }
+
+            defaultGlobal.put("child_registered_count", 2);
+
+            form.put(JsonFormConstants.JSON_FORM_KEY.GLOBAL, defaultGlobal);
+        } catch (JSONException ex) {
+            Timber.e(ex);
+        }
+    }
+
+    public static int getDeliveryDays(String baseEntityId) {
+        Map<String, String> detailMap = PncLibrary.getInstance().getPncRegistrationDetailsRepository().findByBaseEntityId(baseEntityId);
+        if (StringUtils.isNotBlank(detailMap.get(PncConstants.FormGlobalConstants.DELIVERY_DATE))) {
+            LocalDate deliveryDate = LocalDate.parse(detailMap.get(PncConstants.FormGlobalConstants.DELIVERY_DATE), DateTimeFormat.forPattern("dd-MM-yyyy"));
+            return Days.daysBetween(deliveryDate, LocalDate.now()).getDays();
+        } else {
+            return 0;
+        }
+    }
+
+    public static void saveRegistrationFormSilent(String jsonString, RegisterParams registerParams, PncEventActionCallBack callBack) {
+
+        PncRegisterActivityPresenter presenter = new PncRegisterActivityPresenter(null, new PncRegisterActivityModel()) {
+            @Override
+            public void onRegistrationSaved(boolean isEdit) {
+                super.onRegistrationSaved(isEdit);
+                callBack.onPncEventSaved();
+            }
+        };
+
+        presenter.saveForm(jsonString, registerParams);
+
+    }
+
+    public static void saveOutcomeAndVisitFormSilent(String jsonString, Intent data, PncEventActionCallBack callBack) {
+
+        PncRegisterActivityPresenter presenter = new PncRegisterActivityPresenter(null, new PncRegisterActivityModel()) {
+            @Override
+            public void onEventSaved() {
+                super.onEventSaved();
+                callBack.onPncEventSaved();
+            }
+        };
+
+        presenter.savePncForm(jsonString, data);
+
+    }
+
+    public static void processPreChecks(@NonNull String entityId, @NonNull JSONObject jsonForm, @Nullable HashMap<String, String> intentData) {
+        intentData.put(PncDbConstants.KEY.BASE_ENTITY_ID, entityId);
+        PncUtils.addGlobals(entityId, jsonForm);
```

Comment:
Add a filter to this so that it does not run on each form

Reply:
done